src/Core/InstanceConfigTrait.php

```diff
@@ -139,7 +141,11 @@ protected function _configWrite($key, $value, $merge = false)
             } else {
                 $update = [$key => $value];
             }
-            $this->_config = Hash::merge($this->_config, Hash::expand($update));
+            if ($merge === 'shallow') {
+                $this->_config = array_merge($this->_config, Hash::expand($update));
```

Comment:
Don't really think it's necessary. Then we would have a method for no merge / overwriting too.